Sources/Sentry/TracesSampler.m

```diff
@@ -0,0 +1,48 @@
+#import "TracesSampler.h"
+#import "SentryOptions.h"
+#import "SentrySamplingContext.h"
+#import "SentryTransactionContext.h"
+
+@implementation TracesSampler {
+    SentryOptions *_options;
+}
+
+- (instancetype)initWithOptions:(SentryOptions *)options
+{
+    if (self = [super init]) {
+        _options = options;
+        srand48(time(0));
+    }
+    return self;
+}
+
+- (SentrySampleDecision)sample:(SentrySamplingContext *)context
+{
+    if (context.transactionContext.sampled != kSentrySampleDecisionUndecided)
+        return context.transactionContext.sampled;
+
+    if (_options.tracesSampler != nil) {
+        NSNumber *callbackDecision = _options.tracesSampler(context);
+        if (callbackDecision != nil)
+            return [self calcSample:callbackDecision.doubleValue];
+    }
+
+    if (context.transactionContext.parentSampled != kSentrySampleDecisionUndecided)
+        return context.transactionContext.parentSampled;
+
+    if (_options.tracesSampleRate != nil)
+        return [self calcSample:_options.tracesSampleRate.doubleValue];
+
+    return kSentrySampleDecisionNo;
+}
+
+- (SentrySampleDecision)calcSample:(double)rate
+{
+    double r = drand48();
+    if (self.definedRandom != nil)
```

Comment:
I'm an "design for testability" advocate but I'm not sure this is the best way to do this.
What about having `random` be a strategy instead?

Reply:
Yeah, probably a better approach.